torch/distributed/device_mesh.py

```diff
@@ -78,30 +80,61 @@ def get_current_mesh(self) -> "DeviceMesh":
             return self.mesh_stack[-1]
 
         def create_sub_mesh(
-            self, device_mesh: "DeviceMesh", submesh_dim_names: Tuple[str, ...]
+            self,
+            device_mesh: "DeviceMesh",
+            submesh_dim_names: Tuple[str, ...],
+            submesh_dims: List[Tuple[int, ...]],
         ) -> "DeviceMesh":
-            if device_mesh != self.get_root_mesh(device_mesh):
-                raise RuntimeError("Cannot create a submesh from a submesh.")
-
-            # submesh_dims are the mesh dimension of the submesh in the device mesh.
-            submesh_dims = [
-                not_none(device_mesh.mesh_dim_names).index(mesh_dim_name)
-                for mesh_dim_name in submesh_dim_names
-            ]
-            submesh_dim_sizes = [
-                device_mesh.mesh.size(mesh_dim) for mesh_dim in submesh_dims
+            submesh_dim_size = [
+                reduce(
+                    lambda x, y: device_mesh.mesh.size(x) * device_mesh.mesh.size(y),
+                    mesh_dim,
+                )
+                if len(mesh_dim) > 1
+                else device_mesh.mesh.size(mesh_dim[0])
+                for mesh_dim in submesh_dims
```

Comment:
```suggestion
            submesh_dim_size = [
                reduce(
                    lambda x, y: x * y,
                    device_mesh.mesh.size(mesh_dim),
                    initializer=1,
                )
                for mesh_dim in submesh_dims
```

Reply:
I don't think this would work because submesh_dims is a list of tuples. For example, submesh_dims = [(0, 1), (2,), (3, 4)] means we would want to slice out mesh["0_1", "2", "3_4"].

Reply:
in such a case you just need to add another for. But I think it may not be more concise than your original code. We can leave it here.